detekt-rules/src/main/kotlin/org/modelix/detekt/AvoidNonNullAssertionOperator.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.detekt
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import org.jetbrains.kotlin.psi.KtReferenceExpression
+
+class AvoidNonNullAssertionOperator(config: Config) : Rule(config) {
```

Comment:
Interesting how custom rules can be added.

But according to https://github.com/detekt/detekt/issues/4832 it can be replaced by `UnnecessaryNotNullOperator` and `UnsafeCallOnNullableType`

Reply:
UnsafeCallOnNullableType is active by default but doesn't catch all of the cases the custom rule is reporting on. For whatever reason....

Reply:
I just tried it again, even with actively enabling those two rules, despite the fact that they should be active by default, I can't get a single report from them. Not sure if that's a detekt bug.

Reply:
Thist might be realted to type resolution.

See https://github.com/detekt/detekt/issues/2036#issuecomment-543492134
See https://detekt.github.io/detekt/type-resolution.html